src/repr/src/adt/interval.rs

```diff
@@ -376,6 +377,25 @@ impl Interval {
         i128::from(self.micros)
     }
 
+    /// Computes the total number of milliseconds in the interval. Discards fractional millisecond!
+    pub fn as_milliseconds(&self) -> i128 {
+        // unwrap is safe because i32::MAX/i32::MIN number of months will not overflow an i128 when
+        // converted to milliseconds.
+        Self::convert_date_time_unit(
+            DateTimeField::Month,
+            DateTimeField::Milliseconds,
+            i128::from(self.months),
+        ).unwrap() +
+        // unwrap is safe because i32::MAX/i32::MIN number of days will not overflow an i128 when
+        // converted to milliseconds.
+        Self::convert_date_time_unit(
+            DateTimeField::Day,
+            DateTimeField::Milliseconds,
+            i128::from(self.days),
+        ).unwrap() +
+        i128::from(self.micros) / 1000
```

Comment:
Could this all be replaced with 
```Rust
self.as_microseconds() / 1000
```
Or 
```Rust
Self::convert_date_time_unit(
            DateTimeField::Microseconds,
            DateTimeField::Milliseconds,
            self.as_microseconds(),
).unwrap()
```
?

Reply:
Done: replaced with
```
self.as_microseconds() / 1000
```